blog.md

```diff
@@ -1,77 +1,149 @@
+  In this example, in addition to updating the score every 60 seconds, the update will also be triggered when the node or pod resource in the managedcluster changes.
 
-```
-if value >= max
-    score = 100
-else 
-    score = value *100 / max
-```
+  ```go
+  factory.New().WithInformersQueueKeyFunc(
+     func(obj runtime.Object) string {
+        key, _ := cache.MetaNamespaceKeyFunc(obj)
+        return key
+     }, addOnPlacementScoreInformer.Informer()).
+     WithBareInformers(podInformer.Informer(), nodeInformer.Informer()).
+     WithSync(c.sync).ResyncEvery(time.Second*60).ToController("score-agent-controller", recorder)
+  ```
+
+- When should the score be deleted?
+
+### 3. How to calculate the score
 
-Why？
-1. If there is no maximum value, then an infinite interval is mapped to 0-100, and the average mapping cannot be achieved
-2. When the available amount is greater than this maximum value, the cluster can be considered healthy enough to deploy applications
+The score must be in the range -100 to 100, you need to normalize the scores before update it into AddOnPlacementScore. We can divide the normalization of data into two cases according to the specific situation
 
-There may be a better algorithm here, and ideas are welcome.
-## case
-1. 选择出来cpuAvailbale最大的
-2. 选择出来memAvailbale
+- know the actual value of max
+  In this case, it is easy to achieve smooth mapping by formula. Suppose the actual value is X, and X is in the interval [min, max], then``` score ＝ (100 + 100)/(max - min) * (x - min) - 100  ```
 
-## Summary
+- don't know the actual value of max
+
+  But in this case, we often need to set a maximum value ourselves, The reason is that:
+
+  1. If there is no maximum value, then an infinite interval is mapped to 0-100, and the average mapping cannot be achieved
+  2. When the available amount is greater than this maximum value, the cluster can be considered healthy enough to deploy applications.
+
+  So relative to the first case, we need to add extra judgment:
+
+  ```
+  if X >= max
+    score = 100
+  if X <= min 
+    score = -100
+  ```
+
+
+
+In the example, we can't know the maximum value of mem and cpu of the cluster, so use the second calculation method, where ```MAXCPUCOUNT``` and ```MAXMEMCOUNT``` are the values we set manually. At the same time, because the two numbers cpuValue and memValue must be positive numbers, the calculation formula can be simplified:  ```score = 100 / max * X```
+
+```go
+var availableCpu float64
+availableCpu = float64(cpuAllocInt - cpuUsage)
+if availableCpu > MAXCPUCOUNT {
+  cpuValue = int64(100)
+} else {
+  cpuValue = int64(100 / MAXCPUCOUNT * availableCpu)
+}
+
+var availableMem float64
+availableMem = float64((memAllocInt - memUsage) / (1024 * 1024))
+if availableMem > MAXMEMCOUNT {
+  memValue = int64(100)
+} else {
+  memValue = int64(100 / MAXMEMCOUNT * availableMem)
+}
+```
 
-## Reference
+[gif]
```

Comment:
At last, may need an example to show how the pod is running, how the addonplacement score generate. As well as a placement + decesion example.